File: MyMovies.Server/MyMovies.Models/Entities/Movie.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyMovies.Models.Entities
{
    public class Movie
    {
        [Key]
        public Guid ID { get; set; }
        [Required, MaxLength(200)]
        public string Title { get; set; }
        [Column(TypeName = "decimal(4,0)"), MinLength(1900), MaxLength(2100)]
        public decimal? ProductionYear { get; set; }
    }
}

File: MyMovies.Client/src/models/Movie.js
export default class Movie {
  id = "";
  title = "";
  productionYear = null;

  constructor(data) {
    if (data) {
      this.id = data.id ? data.id : "";
      this.title = data.title ? data.title : "";
      this.productionYear = data.productionYear ? data.productionYear : "";
    }
  }
}
